packages/taco/src/taco.ts

```diff
@@ -86,16 +140,47 @@ export const encrypt = async (
  *
  * @throws {Error} If the encryption process throws an error, an error is thrown.
  */
-export const encryptWithPublicKey = async (
+export async function encryptWithPublicKey(
   message: Uint8Array | string,
   condition: Condition,
   dkgPublicKey: DkgPublicKey,
-  authSigner: ethers.Signer,
-): Promise<ThresholdMessageKit> => {
+  authSigner: TacoSigner,
```

Comment:
`TacoSigner` is and internal interface, not external. Otherwise you will force creation of a TACoSigner before calling the API no?

```suggestion
  authSigner: ethers.Signer,
```